tests/invalid_models_tests/test_ordinary_fields.py

```diff
@@ -1226,3 +1227,129 @@ class Model(models.Model):
         msg = f"{expression} cannot be used in db_default."
         expected_error = Error(msg=msg, obj=field, id="fields.E012")
         self.assertEqual(errors, [expected_error])
+
+
+@isolate_apps("invalid_models_tests")
+class GeneratedFieldTests(TestCase):
+    def test_field_not_supported(self):
+        db_persist = (
+            False if not connection.features.supports_stored_generated_columns else True
+        )
+
+        class Model(models.Model):
+            name = models.IntegerField()
+            field = models.GeneratedField(expression=F("name"), db_persist=db_persist)
+
+        expected_errors = []
+        if (
+            not connection.features.supports_stored_generated_columns
+            and not connection.features.supports_virtual_generated_columns
+        ):
+            expected_errors.append(
+                Error(
+                    f"{connection.display_name} does not support GeneratedFields.",
+                    obj=Model._meta.get_field("field"),
+                    id="fields.E220",
+                )
+            )
+        if db_persist and not connection.features.supports_stored_generated_columns:
+            expected_errors.append(
+                Error(
+                    f"{connection.display_name} does not support persisted "
+                    "GeneratedFields.",
+                    obj=Model._meta.get_field("field"),
+                    id="fields.E223",
+                    hint="add the persisted=False argument",
+                )
+            )
```

Comment:
I don't think this block can ever run - I think the `if` condition is always `False`.
```suggestion
```